src/main/java/uk/ac/sanger/sccp/stan/service/PermServiceImp.java

```diff
@@ -200,27 +255,74 @@ public void validatePermValues(Collection<String> problems, Collection<PermData>
         if (!addressesNeither.isEmpty()) {
             problems.add("Neither control type nor time specified for the given address: "+addressesNeither);
         }
+        if (!addressesWithBarcodeNotPosCon.isEmpty()) {
+            problems.add("Control barcode specified for address that is not a positive control: "+addressesWithBarcodeNotPosCon);
+        }
```

Comment:
Do we need to validate here that multiple empty addresses are not assigned with control barcode?